omnigibson/robots/articulated_trunk_robot.py

```diff
@@ -143,21 +47,110 @@ def trunk_control_idx(self):
         """
         return th.tensor([list(self.joints.keys()).index(name) for name in self.trunk_joint_names])
 
+    @property
+    def trunk_action_idx(self):
+        controller_idx = self.controller_order.index("trunk")
+        action_start_idx = sum([self.controllers[self.controller_order[i]].command_dim for i in range(controller_idx)])
+        return th.arange(action_start_idx, action_start_idx + self.controllers["trunk"].command_dim)
+
+    @property
+    def _default_controllers(self):
+        # Always call super first
+        controllers = super()._default_controllers
+
+        # For best generalizability use, joint controller as default
+        controllers["trunk"] = "JointController"
+
+        return controllers
+
+    @property
+    def _default_trunk_ik_controller_config(self):
+        """
+        Returns:
+            dict: Default controller config for an Inverse kinematics controller to control this robot's trunk
+        """
+        return {
+            "name": "InverseKinematicsController",
+            "task_name": "trunk",
+            "control_freq": self._control_freq,
+            "reset_joint_pos": self.reset_joint_pos,
+            "control_limits": self.control_limits,
+            "dof_idx": self.trunk_control_idx,
+            "command_output_limits": (
+                th.tensor([-0.2, -0.2, -0.2, -0.5, -0.5, -0.5]),
+                th.tensor([0.2, 0.2, 0.2, 0.5, 0.5, 0.5]),
+            ),
+            "mode": "pose_delta_ori",
+            "smoothing_filter_size": 2,
+            "workspace_pose_limiter": None,
+        }
+
+    @property
+    def _default_trunk_osc_controller_config(self):
+        """
+        Returns:
+            dict: Default controller config for an Operational Space controller to control this robot's trunk
+        """
+        return {
+            "name": "OperationalSpaceController",
+            "task_name": "trunk",
+            "control_freq": self._control_freq,
+            "reset_joint_pos": self.reset_joint_pos,
+            "control_limits": self.control_limits,
+            "dof_idx": self.trunk_control_idx,
+            "command_output_limits": (
+                th.tensor([-0.2, -0.2, -0.2, -0.5, -0.5, -0.5]),
+                th.tensor([0.2, 0.2, 0.2, 0.5, 0.5, 0.5]),
+            ),
+            "mode": "pose_delta_ori",
+            "workspace_pose_limiter": None,
+        }
+
+    @property
+    def _default_trunk_joint_controller_config(self):
+        """
+        Returns:
+            dict: Default base joint controller config to control this robot's base. Uses velocity
+                control by default.
```

Comment:
Seems to use position control by default

Reply:
fixed